cdb2jdbc/src/main/java/com/bloomberg/comdb2/jdbc/Comdb2ClientInfo.java

```diff
@@ -39,4 +45,54 @@ public static String getCallStack(int layers) {
         }
         return sb.toString();
     }
+    
+    public static String getDriverName() {
+        String name = Driver.class.getPackage().getImplementationTitle();
+        if (name == null) {
+            try { 
+                JarFile jar = new JarFile(Driver.class.getProtectionDomain().getCodeSource().getLocation().getPath());
+                Manifest manifest = new Manifest(jar.getInputStream(jar.getEntry("META-INF/MANIFEST.MF")));
+                Attributes attributes = manifest.getMainAttributes();
+                name = attributes.getValue("Implementation-Title");
+            } catch (IOException e) {
+                logger.log(Level.CONFIG, "Unable to parse driver class manifest");
+            }
+        }
+        return name == null ? "cdb2jdbc" : name;
+    }
+
+    public static String getDriverVersion() {
+        String version = Driver.class.getPackage().getImplementationVersion();
```

Comment:
Does this function already process MANIFEST.MF to get the information?

Reply:
During testing with Maven, `getPackage().getImplementationVersion()` was returning `null` even when the `MANIFEST.MF` existed in the jar it was depending on. I added the check to always try and pull `MANIFEST.MF` from the source jar. Shouldn't be in issue in production but the additional check is safer.

Reply:
I thought you said it returned null because maven would not generate the manifest during `mvn test`. Would it still return null for your integration testcase?

Reply:
Integration tests run *after* `mvn package` through the `failsafe plugin`. The expected test values are pulled directly from `pom.xml` (project.version and project.artifactId) and the actual values returned from the function are compared against that. So unless the version and artifactId in pom.xml are null, the test wouldn't return null.